servemux.go

```diff
@@ -149,6 +149,15 @@ func (s *ServeMux) Handler(r *http.Request) (http.Handler, string) {
 // they would have been executed
 func (s *ServeMux) HandlerAndMiddleware(r *http.Request) (http.Handler, []Middleware, string) {
 
+	path := r.URL.EscapedPath()
+
+	// Check for redirect
+	if path != "/" && strings.HasSuffix(path, "/") {
```

Comment:
This seems very much the same as https://github.com/AndrewBurian/powermux/blob/fc76aac414b613420acdade80c9e2da19cb5745e/servemux.go#L65, except for that it is correctly using `URL.EscapedPath()`. `ServeHTTP` should probably also use `URL.EscapedPath()` and the code can be refactored into one method.